src/rr_pcp.capnp

```diff
@@ -0,0 +1,194 @@
+# rr ReplaySession schema
+
+@0xf55676ebd869d6c1;
+
+using Cxx = import "/capnp/c++.capnp";
+$Cxx.namespace("rr::pcp");
+
+using import "rr_trace.capnp".Registers;
+using import "rr_trace.capnp".ExtraRegisters;
+using import "rr_trace.capnp".Arch;
+using import "rr_trace.capnp".RemoteFd;
+using import "rr_trace.capnp".CString;
+using import "rr_trace.capnp".Device;
+using import "rr_trace.capnp".Inode;
+using import "rr_trace.capnp".RemotePtr;
+using import "rr_trace.capnp".FrameTime;
+using import "rr_trace.capnp".Tid;
+using import "rr_trace.capnp".Fd;
+using import "rr_trace.capnp".Path;
+using import "rr_trace.capnp".Ticks;
+
+
+using FileMonitorType = Int32;
+struct FileMonitor {
+  fd @0 :Fd;
+  type @1 :FileMonitorType;
+  union {
+    mmap :group {
+      dead @2 :Bool;
+      device @3 :Device;
+      inode @4 :Inode;
+    }
+    procFd :group {
+      tid @5 :Tid;
+      serial @6 :UInt32;
+    }
+    procMem :group {
+      tid @7 :Tid;
+      serial @8 :UInt32;
+      execCount @9 :UInt32;
+    }
+    stdio @10 :Fd;
+    procStat @11 :Data;
+    bpf :group {
+      keySize @12: UInt64;
+      valueSize @13 :UInt64;
+    }
+  }
+}
+
+struct KernelMapping {
+  start @0 :RemotePtr;
+  end @1 :RemotePtr;
+  fsname @2 :CString;
+  device @3 :Device;
+  inode @4 :Inode;
+  protection @5 :Int32;
+  flags @6 :Int32;
+  offset @7 :UInt64;
+  mapType :union {
+    file :group { # mapping of a file
+      skipMonitoringMappedFd @8 :Bool; # unsure if needed, or how it's used
```

Comment:
This is used to communicate the decision about whether or not to monitor an mmaped file during recording (see `monitor_this_fd` in record_syscall.cc's `process_mmap`). Whether or not there's a monitor affects the behavior of the syscallbuf so it has to be the same in recording and replay.

I don't think you need this for PCPs because you are separately restoring the state of the FdTable.